rcl_lifecycle/src/com_interface.c

```diff
@@ -250,9 +250,9 @@ rcl_lifecycle_com_interface_publish_notification(
   rcl_lifecycle_com_interface_t * com_interface,
   const rcl_lifecycle_state_t * start, const rcl_lifecycle_state_t * goal)
 {
-  msg.start_state.id = start->id;
+  msg.start_state.id = (uint8_t) start->id;
```

Comment:
So, turning on the warning here has shown a potential bug (which is a good thing!).

The problem is that rcl_lifecycle_state_t has an `id` member of type `unsigned int`: https://github.com/ros2/rcl/blob/0ca8c6162136594e490ba69fbf288de1b5f7073d/rcl_lifecycle/include/rcl_lifecycle/data_types.h#L35

But the message we are trying to stuff it into (State message) has an `id` member of type `uint8`: https://github.com/ros2/rcl_interfaces/blob/7a6ebaa2064151a48dd189cd514fc7db5d260047/lifecycle_msgs/msg/State.msg#L54

That means that it is conceivable that we could generate an ID internally that we cannot represent in the message.  Just think of the case where we have 256 as the `start->id`; this will wrap back to 0 and publish a bogus state id.

There are 2 ways we can fix this:
1.  Increase the size of the State.msg field to accommodate the possible ID.
1.  Make the internal `rcl_lifecycle_state_t` member `id` be a uint8.

I would go for the second one.  Given that these are meant to be the IDs of states, then 255 really should be enough.  If it turns out it isn't in the future, we'll have to do work on both the internal structure and the external message to expand them.